packages/features/ee/teams/components/MemberInvitationModal.tsx

```diff
@@ -1,127 +1,137 @@
-      role: target.elements["role"].value,
-      usernameOrEmail: target.elements["inviteUser"].value,
-      sendEmailInvitation: target.elements["sendInviteEmail"].checked,
-    });
-  }
+  const validateUniqueInvite = (value: string) => {
+    return !(
+      props.members.some((member) => member?.username === value) ||
+      props.members.some((member) => member?.email === value)
+    );
+  };
 
   return (
-    <Dialog open={props.isOpen} onOpenChange={props.onExit}>
+    <Dialog
+      open={props.isOpen}
+      onOpenChange={() => {
+        props.onExit();
+        newMemberFormMethods.reset();
+      }}>
       <DialogContent
         type="creation"
         useOwnActionButtons
         title={t("invite_new_member")}
         description={
           <span className=" text-sm leading-tight text-gray-500">
-            Note: This will <span className="font-medium text-gray-900">cost an extra seat ($12/m)</span> on
-            your subscription if this invitee does not have a TEAM account.
+            Note: This will <span className="font-medium text-gray-900">cost an extra seat ($15/m)</span> on
+            your subscription.
           </span>
         }>
-        <form onSubmit={inviteMember}>
+        <Form form={newMemberFormMethods} handleSubmit={(values) => props.onSubmit(values)}>
           <div className="space-y-4">
-            <TextField
-              label={t("email_or_username")}
-              id="inviteUser"
-              name="inviteUser"
-              placeholder="email@example.com"
-              required
+            <Controller
+              name="emailOrUsername"
+              control={newMemberFormMethods.control}
+              rules={{
+                required: t("enter_email_or_username"),
+                validate: (value) => validateUniqueInvite(value) || t("member_already_invited"),
+              }}
+              render={({ field: { onChange }, fieldState: { error } }) => (
+                <>
+                  <TextField
+                    label={t("email_or_username")}
+                    id="inviteUser"
+                    name="inviteUser"
+                    placeholder="email@example.com"
+                    required
+                    onChange={onChange}
+                  />
+                  {error && <span className="text-sm text-red-800">{error.message}</span>}
+                </>
+              )}
+            />
+            <Controller
+              name="role"
+              control={newMemberFormMethods.control}
+              defaultValue={options[0]}
+              render={({ field: { onChange } }) => (
+                <div>
+                  <label
+                    className="mb-1 block text-sm font-medium tracking-wide text-gray-700"
+                    htmlFor="role">
+                    {t("role")}
+                  </label>
+                  <Select
+                    defaultValue={options[0]}
+                    options={options.slice(0, 2)}
```

Comment:
Would it be safer and perhaps clearer if you'd do something like `options.filter(option => option.value !== 'OWNER')`? This prevents owner appearing in the options by accident if someone where to change the order of the array (something people wouldn't expect when changing the order). Also makes it immediately clear what you're doing here.